### ISSUE TITLE
Load Homepage with PJAX

### ISSUE THREAD

**Issue by jorditost** [chrisguitarguy/pjaxy#2]
Hi Chris,

It seems that the homepage is always loaded by reloading the whole page, but I'd be interesting to load it with PJAX.

I have inside the folder 'pjax-templates' also a home.php with its content.

Is there a way to fix?

Thanks!

**Comment by chrisguitarguy** [comment]
Can you post the content of your `home.php` pjax template?

**Comment by jorditost** [comment]
Here the code:

``` php
<?php get_pjaxy_page_info(); ?>

<title><?php bloginfo('name'); echo ' | '; bloginfo('description'); ?></title>

<?php 
    $page = get_page_by_path('home');
    if ($page) :
?>  
    <div id="page-home" class="content inside" data-section="home">
        <h1 id="home-claim"><?php echo __($page->post_excerpt); ?></h1>

        <div class="entry">
            <?php echo format_content($page->post_content); ?>
        </div>
    </div><!-- .content -->

    <div id="home-gallery">
        <ul>
            <?php echo get_post_gallery_by_ID($page->ID, 'full', false, 'li', true); ?>
        </ul>
    <?php
        global $isMobile;
        if ( !$isMobile ) { ?>
        <div id="gallery-drag-help"></div>
    <?php } ?>
    </div><!-- #intro-gallery -->

<?php endif; ?>
```

**Comment by jorditost** [comment]
Hi Chris,

I've been testing and it seems that the _home.php_ template is not working at all. In _pjax.jquery.js_, PJAX is getting the title and other parameters correctly, but no content is read. Look at this piece of code (line 200). With _home.php_ it reloads the page and returns:

``` php
    if (!container.contents) {
      window.location = container.url
      return
    }
```

Just by changing the name of the template from _home.php_ to _index.php_, it works properly.

**Comment by chrisguitarguy** [comment]
Yes, there's an issue with it not finding the home page.

Is this meant to be a static home page?  Eg. in Options > Read "Front Page Shows" is set to a static page?

**Comment by jorditost** [comment]
No, it's not statically defined. It loads the page I query directly in home.php.

**Comment by chrisguitarguy** [comment]
I'm confused what you're trying to do, but Pjaxy will follow the WordPress template hierarchy.  Without seeing your entire theme, I can't help you.

In other words: if this is the home page, WordPress will load `home.php` if you have the front page set to be a posts page.  Otherwise, it's going to lood for `front-page.php`, then page templates, then `page.php` and finally back to `index.php`.

**Comment by ChaplinWang** [comment]
found one issue in pjaxy.js, the init call to pjax does not match current pjax init signature.

I have created a pull request to fix this: https://github.com/chrisguitarguy/pjaxy/pull/4
